src/main/java/calculator/model/Calculator.java

```diff
@@ -0,0 +1,50 @@
+package calculator.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Calculator {
+
+    private static final char FIRST_DEFAULT_DELIMITER = ':';
+    private static final char SECOND_DEFAULT_DELIMITER = ',';
+
+    private static Calculator calculator;
+    private ArrayList<Character> delimiters = new ArrayList<>(Arrays.asList(FIRST_DEFAULT_DELIMITER, SECOND_DEFAULT_DELIMITER));
+    private ArrayList<Integer> operands;
+    private int result;
+
+    private Calculator() {
+
+    }
+
+    public static Calculator getInstance() {
+        if (calculator == null) {
+            calculator = new Calculator();
+        }
+        return calculator;
+    }
+
+    public void addCustomDelimiter(char newDelimiter) {
+        delimiters.add(newDelimiter);
+    }
+
+    public ArrayList<Character> getDelimiters() {
+        return delimiters;
+    }
+
+    public void setOperands(ArrayList<Integer> operands) {
+        this.operands = operands;
+    }
+
+    public void saveResult() {
+        int sum = 0;
+        for (Integer operand : operands) {
+            sum += operand;
+        }
+        result = sum;
+    }
```

Comment:
멤버변수롤 result를 유지하고 그것을 save하는 형태로 구현하신 이유가 있으실까요!?
getResult를 하기 전에 saveResult를 불러줘야 올바른 값을 얻을 수 있다는 점이 사용하는 입장에서 실수를 유발할 수 있을 것 같습니다! 이런 형태는 어떨까요?
```suggestion
    public int calculate() {
        int sum = 0;
        for (Integer operand : operands) {
            sum += operand;
        }
        result = sum;
    }
```

Reply:
저는 "model 내부에서 계산을 통해 result를 model에 저장한다"라는 행위와, "model 외부에서 멤버 변수인 result를 얻기 위해 접근한다"라는 행위를 구분했습니다! 그런데 **호출 순서가 명확히 존재해서 이 부분에서 실수를 유발할 수도 있다**는 생각은 못해봤네요.. 좋은 지적 감사합니다! 👍